low-level/framework/utils/iem.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+import syslog
+import os
+
+from framework.base.sspl_constants import IEM_DATA_PATH
+from framework.utils.service_logging import logger
+
+class Iem:
+    Severity = {
+        "INFO" : "I", "WARN" : "W", "ERROR" : "E", "CRITICAL" : "C" }
+
+    # EVENT_CODE = [event_code, event]
+    EVENT_CODE = {
+        "IPMITOOL_ERROR" : ["0050010001", "IPMITOOL"],
+        "IPMITOOL_AVAILABLE" : ["0050010002", "IPMITOOL"],
+        "HDPARM_ERROR" : ["0050010003", "HDPARM"],
+        "HDPARM_AVAILABLE" : ["0050010004", "HDPARM"],
+        "SMARTMONTOOL_ERROR" : ["0050010005", "SMARTMONTOOL"],
+        "SMARTMONTOOL_AVAILABLE" : ["0050010006", "SMARTMONTOOL"],
+        "UDISKS2_UNAVAILABLE" : ["0050010007", "UDISKS2"],
+        "UDISKS2_AVAILABLE" : ["0050010008", "UDISKS2"],
+        "KAFKA_NOT_ACTIVE" : ["0050020001", "KAFKA"],
+        "KAFKA_ACTIVE" : ["0050020002", "KAFKA"]
+    }
+
+    # EVENT_STRING = { event_code : [description, impact, recommendation] }
+    EVENT_STRING = {
+        "0050010001" : ["Ipmitool command execution error.",
+            "Server resource monitoring halted.",
+            "Reinstall/reconfigure ipmitool package."],
+        "0050010002" : ["Ipmitool command executed successfully.",
+            "Server resource monitoring started.",
+            ""],
+        "0050010003" : ["Hdparm command execution error.",
+            "Server local drives can not be monitored.",
+            "Reinstall/reconfigure hdparm package."],
+        "0050010004" : ["Hdparm command executed susscessfully.",
+            "Started server local drives monitoring.",
+            ""],
+        "0050010005" : ["Smartctl command execution error.",
+            "Can not fetch drive information.",
```

Comment:
Unable to fetch server drive SMART test results & related health info.

Reply:
Done.